app/decorators/user_decorator.rb

```diff
@@ -1,9 +1,9 @@
 class UserDecorator < ApplicationDecorator
   delegate_all
 
-  def proposal_url(proposal)
+  def proposal_url(proposal, force_public_url=false)
     event = proposal.event
-    if model.staff_for? event
+    if model.staff_for?(event) && !force_public_url
```

Comment:
Hmm, okay.  Maybe we should pass in the proposal_url into the create_for method since the notify method knows which one it should be?